src/diag.rs

```diff
@@ -187,7 +187,7 @@ fn annotate_diagnostic(notes: &mut Vec<Notation>,
     }
 
     fn t(v: &Token) -> String {
-        String::from_utf8((**v).to_owned()).expect("utf-8 is checked before making Token")
+        unsafe { String::from_utf8_unchecked((**v).to_owned()) }
```

Comment:
I thought all the `Token` objects are created during parsing, and we certainly are checking for ASCII there. Anything which comes from the file buffer is checked ASCII (although it may be hard to make the checking happen at exactly `new()` without a performance hit, so maybe the parser gets a pass on the checking via some `new_unchecked()`).

Currently, as I said this is only changed for consistency with `as_str`. You should focus on that, which does get called a lot. In particular, exporting a large proof will cause many many calls to `as_str` (although since I do lookup beforehand, it's proportional to the number of compressed steps, not uncompressed, when outputting in `Normal` style).

This function should probably be an `as_string` function in `parser` or `utils`, but since you made a local function I assumed you were using it a lot here so I didn't try to factor it out.